containers/daap-create-schema/src/var/task/create_schema.py

```diff
@@ -0,0 +1,153 @@
+            "base_image_version": os.getenv("BASE_VERSION", "unknown"),
+            "data_product_name": data_product_name,
+            "table_name": table_name,
+        }
+    )
+
+    logger.info(f"event: {event}")
+
+    request_body = json.loads(event.get("body"))
+
+    http_method = event.get("httpMethod")
+
+    if not http_method == HTTPMethod.POST:
+        error_message = f"Sorry, {http_method} isn't allowed."
+        logger.error(f"error message: {error_message}, input: {event}")
+        return response_status_400(error_message)
+
+    schema = DataProductSchema(
+        data_product_name=data_product_name,
+        table_name=table_name,
+        logger=logger,
+        input_data=request_body.get("schema"),
+    )
+
+    if schema.exists:
+        error_msg = (
+            f"v1 of this schema for table {table_name} already exists. You can upversion this schema if "
+            "there are changes from v1 using the PUT method of this endpoint. Or if this is a different "
+            "table then please choose a different name for it."
+        )
+        logger.error("create schema called where v1 already exists.")
+        return response_status_403(error_msg)
+
+    if not schema.has_registered_data_product:
+        error_msg = (
+            f"Schema for {table_name} has no registered metadata for the data product it belongs to. "
+            "Please first register the data product metadata using the POST method of the /data-product/register"
+            " endpoint."
+        )
+        logger.error("schema has no associated registered data product metadata.")
+        return response_status_403(error_msg)
+
+    # Code below that handles verisoning of a data product will be moved to a central module eventually.
+
+    # if schema already exist then we need to minor version increment to dataproduct metadata and schema
+    if schema.valid:
+        schema.convert_schema_to_glue_table_input_csv()
+        if not schema.parent_product_has_registered_schema:
+            metadata_dict = schema.parent_data_product_metadata
+            metadata_dict["schemas"] = [table_name]
+
+            # write v1.0 of metadata updated with registered schema, this is the only time we overwrite v1
+            DataProductMetadata(
+                data_product_name=data_product_name,
+                logger=logger,
+                input_data=metadata_dict,
+            ).write_json_to_s3(
+                DataProductConfig(data_product_name).metadata_path("v1.0").key
+            )
+            schema.write_json_to_s3(
+                DataProductConfig(data_product_name).schema_path(table_name, "v1.0").key
+            )
+            msg = f"Schema for {table_name} has been created in the {data_product_name} data product"
+            logger.info("Schema successfully created")
+            return response_status_200(msg)
+        else:
+            # write to next minor version increment
```

Comment:
Just want to reiterate the logic 

 When there is a schema already and new schema is added , minor version  increment happens 
1. New version is generated  for schema
2. New folder created 
3. Data product metadata is copied to it se folder
4. New schema is added to the s3 folder

Reply:
Does   `  s3_copy_folder_to_new_folder` copies other schema to the new version folder ?

Reply:
Yeah so the `DataProductSchema` class checks whether any schema have already been registered in its parent data product. That property is used to know whether to add the schema to the current version of the metadata and save schema in v1.0  folder, or to copy latest version folder's contents to next minor version folder, append schema name to the data product metadata and save schema and metadata files in that next version folder.

If you look at the old products-development bucket in the test_product folder you'll see some different version folders i created when testing (this will go in the metadata bucket now but it didn't exist at the time)